server/tls/tls_certificate_retriever.go

```diff
@@ -8,8 +8,11 @@ import (
 )
 
 // TlsCertificateRetriever is the interface used by
-// both windows cert store implementation, linux and cert from file retriever.
+// both windows and linux and cert from file retriever.
 type TlsCertificateRetriever interface {
 	GetCertificate() (*x509.Certificate, error)
 	GetPrivateKey() (crypto.PrivateKey, error)
+	ReadFile() ([]byte, error)
+	Decrypt([]byte)  (string, error)
+	ParsePEMFile([]byte) (error)
```

Comment:
It seems the new interface methods are unused, and can remain internal to the implementations.

Reply:
Yeah, good point, I removed it from the interface.